src/views/News/index.js

```diff
@@ -146,31 +256,205 @@ export default class StudentNews extends Component {
       currentlyEditing: newsID,
     });
+
+    /*
+    Error checking. Theoretically, this code is designed so that
+    When the get method in the API service is called from the frontend,
+    it will return the image data, even though that's not the value
+    of the image column in news entries. But in the impossible event that
+    it somhow DID return the path of the image instead of the image data,
+    not only would that produce garbage and make the cropper have trouble,
+    but it also is a potential security concern; it sends data back to the
+    client that shouldn't be sent anywhere.
+    */
+    let base64Test = /^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$/;
+
+    if (base64Test.test(newsItem.Image) && newsItem.Image !== null) {
+      let newImageData = 'data:image/jpg;base64,' + newsItem.Image;
+      this.setState({ showCropper: newImageData, justShowPicture: true });
+    } else {
+      this.setState({ showCropper: null });
+    }
   }
 
-  updateSnackbar(message) {
-    this.setState({ snackbarMessage: message });
-    this.setState({ snackbarOpen: true });
+  /**
+   * Handles closing the Photo Updater Dialog Box
+   */
+  async handleCloseCancel() {
+    this.setOpenPhotoDialog(false);
+    this.setShowCropper(null);
+    this.clearPhotoDialogErrorTimeout();
+  }
+
+  onCropperZoom(event) {
+    if (event.detail.ratio > 1) {
+      event.preventDefault();
+      React.cropperRef.current.cropper.zoomTo(1);
+    }
+  }
+
+  /**
+   * Handles the acceptance of the user dropping an image in the Photo Uploader in News submission
+   *
+   * @param {*} fileList The image dropped in the Dropzone of the Photo Uploader
+   */
+  onDropAccepted = (fileList) => {
+    var previewImageFile = fileList[0];
+    var reader = new FileReader();
+    reader.onload = () => {
+      this.imageOnLoadHelper(reader);
+    };
+    reader.readAsDataURL(previewImageFile);
+  };
+
+  /**
+   * Handles the rejection of the user dropping an invalid file in the Photo Updater Dialog Box
+   * Copied from Identification
+   */
+  async onDropRejected() {
+    await this.clearPhotoDialogErrorTimeout();
+    this.setPhotoDialogError('Sorry, invalid image file! Only PNG and JPEG images are accepted.');
+    this.setState({ justShowPicture: false });
   }
 
   // Handles the actual 'edit' submission
   async handleUpdate() {
     let newsID = this.state.currentlyEditing;
+
+    let imageData;
+
+    if (this.cropperRef.current === null && this.state.showCropper === null) {
+      imageData = null;
+    } else if (this.cropperRef.current === null && this.state.showCropper !== null) {
+      imageData = this.state.showCropper.replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
+    } else if (this.cropperRef.current !== null) {
+      imageData = this.cropperRef.current.cropper
+        .getCroppedCanvas({ width: this.CROP_DIM })
+        .toDataURL()
+        .replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
+    }
```

Comment:
You could simplify the logic here a decent bit:
```suggestion
    let imageData = null;

    if (this.cropperRef.current !== null) {
       imageData = this.cropperRef.current.cropper
        .getCroppedCanvas({ width: this.CROP_DIM })
        .toDataURL()
        .replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
    } else if (this.state.showCropper !== null) {
      imageData = this.state.showCropper.replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
    }
```

Reply:
Oh yes, this is much better, thanks!